src/Init/Init_FFTW.cpp

```diff
@@ -363,6 +385,15 @@ void Patch2Slab( real *VarS, real *SendBuf_Var, real *RecvBuf_Var, long *SendBuf
       TempBuf_SIdx   [r] = (long*)malloc( MemSize[r]*sizeof(long)        );
       TempBuf_Var    [r] = (real*)malloc( MemSize[r]*sizeof(real)*PSSize );
       List_NSend_SIdx[r] = 0;
+
+      if ( List_PID[r] == NULL )
+          Aux_Error( ERROR_INFO, "List_PID[%d] is NULL on Rank %d !!\n", r, MPI_Rank );
+      if ( List_k[r] == NULL )
+          Aux_Error( ERROR_INFO, "List_k[%d] is NULL on Rank %d !!\n", r, MPI_Rank );
+      if ( TempBuf_SIdx[r] == NULL )
+          Aux_Error( ERROR_INFO, "TempBuf_SIdx[%d] is NULL on Rank %d !!\n", r, MPI_Rank );
+      if ( TempBuf_Var[r] == NULL )
+          Aux_Error( ERROR_INFO, "TempBuf_Var[%d] is NULL on Rank %d !!\n", r, MPI_Rank ); 
```

Comment:
Maybe can apply the same check for `List_PID`, `List_k`, `TempBuf_SIdx` and `TempBuf_Var` for pointers returned by `realloc` below as well~?